### ISSUE TITLE
`expo ios` loops forever

### ISSUE THREAD

**Issue by raphaelrk** [expo/expo-cli#185]
## Description

Having trouble installing expo or running my expo projects on iOS. Whether I run `expo ios`, `expo install:ios`, or press `i` when running `expo start`, and whether or not I already have expo installed on my simulator, this results in a loop between xcodebuild, osascript, node, etc. and doesn't install expo on my phone or open my project in the simulator.

## Expected Behavior

- Expo is installed on device
- My app is opened in expo

## Observed Behavior

- Terminal loops between a few things indefinitely
- Computer heats up
- Expo is not installed / app does not open
- Still having issue after restarting computer

Gif of the issue (the programs it loops between varies -- for example this gif includes ngrok but that doesn't always happen): [gif](https://gfycat.com/HairyAdolescentAppaloosa)

[Expo forums post](https://forums.expo.io/t/cant-open-in-ios-simulator/15885/2)

[Possibly the same issue in the react-native repo](https://github.com/facebook/react-native/issues/16487)

## Environment

```
  Expo CLI 2.4.0 environment info:
    System:
      OS: macOS High Sierra 10.13.6
      Shell: 3.2.57 - /bin/sh
    Binaries:
      Node: 11.1.0 - /usr/local/bin/node
      Yarn: 1.12.3 - /usr/local/bin/yarn
      npm: 6.4.1 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    IDEs:
      Xcode: 10.1/10B61 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^31.0.4 => 31.0.5 
      react: 16.5.0 => 16.5.0 
      react-native: https://github.com/expo/react-native/archive/sdk-31.0.1.tar.gz => 0.57.1 
      react-navigation: ^3.0.0-rc.0 => 3.0.0-rc.5 
    npmGlobalPackages:
      expo-cli: 2.4.0

  Diagnostics report:
    https://exp-xde-diagnostics.s3.amazonaws.com/raphaelrk-3cef6c11-8f01-4492-b0fa-b745b7cbe758.tar.gz
```

Also specify:

- Phone/emulator/simulator platform, model and version: was using iOS 11 simulators then switched to iOS 12 simulators (updated xcode). Issue has occurred on both OS's and across simulators.

Some other details that may/may not be relevant:
- I've recently only been able to run expo commands with sudo
- Recently ran [this](https://github.com/kmagiera/react-native-gesture-handler) with `react-native run-ios`, a command I hadn't previously manually run. Don't think I had issues before that. But also don't know why that would cause any issues.

**Comment by fson** [comment]
Can you try to reset your simulator? The fastest way to do this is to quit the Simulator.app and run this command in the terminal:
```
xcrun simctl erase all
```
(Note that this will **erase all content and settings** in all your simulator devices.)

After this try `expo start --ios` again.  If it *still* doesn't work, I'd open Xcode and see if it wants you to accept a license agreement or to install command line tools.

Please let me know if this helped!

**Comment by raphaelrk** [comment]
Thank you @fson - tried what you said but still running into the same issue

**Comment by fson** [comment]
> I've recently only been able to run expo commands with sudo

This is not normal. Sounds like something might be wrong with file permissions. What is the error you get when you try to run `expo start` without sudo?

**Comment by raphaelrk** [comment]
Wow, the weirdest things just happened and it's fixed!

So I was still running into this issue. I was running `sudo expo install:ios` and this was being output to the console:
```
[16:07:29] Uninstalling Expo from iOS simulator.
[16:07:30] Downloading latest version of Expo
[16:07:31] Installing Expo on iOS simulator
[16:07:33] Done!
```
But it wasn't actually getting installed, so I went to this issue to look for any new updates. I saw your suggestion and tried `expo start` without sudo and it told me to check `~/.expo/log` (which is what it's been doing).

`~/.expo/log` was really long and seemed to have a lot of unrelated issues, so I figured if I deleted it I'd get a fresh log with just the relevant issue. So I run `rm ~/.expo/log` and then `expo install:ios` and everything started working again! I was able to run `expo start` normally, and then press `i` and it opened the simulator with my project.
